### PR TITLE
Bug/csi 1888 k8s code generation fail

### PR DESCRIPTION
**Pull request by matancarmeli7**

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -142,7 +144,6 @@ func main() {
 	if err = serveCRMetrics(cfg); err != nil {
 		log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
 	}
-
```
Comment (oriyarde) [inline cmd/manager/main.go:145]: could you undo these while you're at it?

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go [hunk 2]
```diff
@@ -661,3 +662,33 @@ func (r *ReconcileIBMBlockCSI) getClusterRoleBindings(instance *ibmblockcsi.IBMB
...
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileIBMBlockCSI) getCSIDriver(instance *ibmblockcsi.IBMBlockCSI) *storagev1.CSIDriver {
+	csiDriver := instance.GenerateCSIDriver()
+
+	return csiDriver
+}
```
Comment (oriyarde) [inline pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:694]: missing newline at EOF

Comment (oriyarde) [inline pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:694]: I don't think this function is necessary

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go [hunk 3]
```diff
@@ -216,6 +216,10 @@ func (r *ReconcileIBMBlockCSI) Reconcile(request reconcile.Request) (reconcile.R
 			return reconcile.Result{}, err
 		}
 
+		if err := r.deleteCSIDriver(instance); err != nil {
+			return reconcile.Result{}, err
+		}
```
Comment (oriyarde) [inline pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:221]: does this cleanup work for all upgrade scenarios on all currently supported k8s versions?

File: pkg/internal/ibmblockcsi/static_resource_generator.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ import (
 	"github.com/IBM/ibm-block-csi-operator/pkg/util/boolptr"
 	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
-	storagev1beta1 "k8s.io/api/storage/v1beta1"
+	storagev1 "k8s.io/api/storage/v1"
```
Comment (oriyarde) [inline pkg/internal/ibmblockcsi/static_resource_generator.go:24]: could you assign CSI-2415 to yourself while you're at it?

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go (hunk 3 above)
Comment (matancarmeli7) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:221]: yes

File: cmd/manager/main.go (hunk 1 above)
Comment (oriyarde) [reply cmd/manager/main.go:145]: I don't see any change here

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go (hunk 2 above)
Comment (oriyarde) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go]: now there is an extra line...

Comment (matancarmeli7) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go]: done

File: cmd/manager/main.go (hunk 1 above)
Comment (matancarmeli7) [reply cmd/manager/main.go]: dene

Comment (oriyarde) [reply cmd/manager/main.go]: I still see this one:
image

**Review by oriyarde — APPROVED** [review]
nice job!

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go [hunk 5]
```diff
@@ -661,3 +662,27 @@ func (r *ReconcileIBMBlockCSI) getClusterRoleBindings(instance *ibmblockcsi.IBMB
...
+	} else if err != nil {
+		logger.Error(err, "failed to get CSIDriver", "Name", csiDriver.GetName())
+		return err
+	} else {
+		logger.Info("deleting CSIDriver", "Name", csiDriver.GetName())
+		if err := r.client.Delete(context.TODO(), found); err != nil {
+			logger.Error(err, "failed to delete CSIDriver", "Name", csiDriver.GetName())
+			return err
+		}
+	}
```
Comment (zingero) [inline pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:686]: We can simplify this code and remove the double negative:
```suggestion
	if err == nil {
        	logger.Info("deleting CSIDriver", "Name", csiDriver.GetName())
		if err := r.client.Delete(context.TODO(), found); err != nil {
			logger.Error(err, "failed to delete CSIDriver", "Name", csiDriver.GetName())
			return err
		}
	 } else if errors.IsNotFound(err) {
		return nil
	} else {
		logger.Error(err, "failed to get CSIDriver", "Name", csiDriver.GetName())
		return err
	}
```

Comment (oriyarde) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:686]: whatever you decide - note that this code structure exists in several places

Comment (matancarmeli7) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go]: done

Comment (oriyarde) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go]: could you change [said places](https://github.com/IBM/ibm-block-csi-operator/pull/154#discussion_r642520537) while you're at it?

**Review by zingero — APPROVED** [review]

**Pull request merged**